src/mgmt/srv_query.c

```diff
@@ -852,30 +846,16 @@ ds_mgmt_dev_replace(uuid_t old_dev_uuid, uuid_t new_dev_uuid, Ctl__DevManageResp
 		DP_UUID(old_dev_uuid), DP_UUID(new_dev_uuid));
 
 	D_ALLOC(resp->device->uuid, DAOS_UUID_STR_SIZE);
-	if (resp->device->uuid == NULL) {
-		rc = -DER_NOMEM;
-		goto out;
-	}
+	if (resp->device->uuid == NULL)
+		return -DER_NOMEM;
 	uuid_unparse_lower(new_dev_uuid, resp->device->uuid);
 
 	uuid_copy(replace_dev_info.old_dev, old_dev_uuid);
 	uuid_copy(replace_dev_info.new_dev, new_dev_uuid);
 	rc = dss_ult_execute(bio_storage_dev_replace, &replace_dev_info, NULL, NULL,
 			     init_xs_type(), 0, 0);
-	if (rc != 0) {
+	if (rc != 0)
 		DL_ERROR(rc, "ULT did not complete storage_dev_replace");
-		/* BIO device state after unsuccessful reintegration */
-		// resp->device->dev_state = CTL__NVME_DEV_STATE__EVICTED;
-		goto out;
-	}
-
-	/* BIO device state after successful reintegration */
-	// resp->device->dev_state = CTL__NVME_DEV_STATE__NORMAL;
-out:
-	if (rc != 0) {
-		if (resp->device->uuid != NULL)
-			D_FREE(resp->device->uuid);
-	}
```

Comment:
Don't need to free in the error case anymore?

Reply:
`drpc_dev_manage_free()` frees resp->device after `ds_mgmt_dev_replace()` is called from srv_drpc.c